hbase/hbase/stackable/patches/2.6.1/0006-Update-property-usage-for-bound-ports.patch

```diff
@@ -0,0 +1,174 @@
+     int port = this.serverManager.getInfoPort(sn);
+     return port == 0
+-      ? conf.getInt(HConstants.REGIONSERVER_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT)
++      ? conf.getInt(HConstants.REGIONSERVER_BOUND_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT)
+       : port;
+   }
+ 
+diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+index 11bd1e58b5..abeeab3552 100644
+--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
++++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegionServer.java
+@@ -25,6 +25,7 @@ import static org.apache.hadoop.hbase.HConstants.DEFAULT_SLOW_LOG_SYS_TABLE_CHOR
+ import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_COORDINATED_BY_ZK;
+ import static org.apache.hadoop.hbase.HConstants.HBASE_SPLIT_WAL_MAX_SPLITTER;
+ import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_BOUND_INFO_PORT;
++import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_INFO_PORT;
+ import static org.apache.hadoop.hbase.HConstants.REGIONSERVER_PORT;
+ import static org.apache.hadoop.hbase.HConstants.RPC_CLIENT_BIND_ADDRESS;
+ import static org.apache.hadoop.hbase.master.waleventtracker.WALEventTrackerTableCreator.WAL_EVENT_TRACKER_ENABLED_DEFAULT;
+@@ -791,7 +792,7 @@ public class HRegionServer extends Thread
+   }
+ 
+   protected int getUseThisInfoPortInstead(Configuration conf) {
+-    int port = conf.getInt(REGIONSERVER_BOUND_INFO_PORT, 0);
++    int port = conf.getInt(REGIONSERVER_INFO_PORT, 0);
+     return port != 0 ? port : this.infoServer != null ? this.infoServer.getPort() : -1;
+   }
+ 
+@@ -2463,11 +2464,11 @@ public class HRegionServer extends Thread
+    */
+   private void putUpWebUI() throws IOException {
+     int port =
+-      this.conf.getInt(HConstants.REGIONSERVER_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
++      this.conf.getInt(REGIONSERVER_BOUND_INFO_PORT, HConstants.DEFAULT_REGIONSERVER_INFOPORT);
+     String addr = this.conf.get("hbase.regionserver.info.bindAddress", "0.0.0.0");
+ 
+     if (this instanceof HMaster) {
+-      port = conf.getInt(HConstants.MASTER_INFO_PORT, HConstants.DEFAULT_MASTER_INFOPORT);
++      port = conf.getInt(HConstants.MASTER_BOUND_INFO_PORT, HConstants.DEFAULT_MASTER_INFOPORT);
+       addr = this.conf.get("hbase.master.info.bindAddress", "0.0.0.0");
+     }
+     // -1 is for disabling info server
+@@ -2503,12 +2504,18 @@ public class HRegionServer extends Thread
+         LOG.info("Retry starting http info server with port: " + port);
+       }
+     }
+-    port = useThisInfoPortInstead;
+-    conf.setInt(HConstants.REGIONSERVER_INFO_PORT, port);
++
++    // update bound ports
++    port = this.infoServer.getPort();
++    conf.setInt(REGIONSERVER_BOUND_INFO_PORT, port);
++    conf.setInt(HConstants.MASTER_BOUND_INFO_PORT, port);
```

Comment:
Is anything actually reading these new values?

Reply:
I'm not 100% sure, but the original code wrote back the port (which may have been incremented in the preceding loop for auto-bind cases) to the conf, so I've retained that. And the original code didn't make a distinction between bound and advertised, so I've written both back.